src/libsystemd-network/sd-radv.c

```diff
@@ -546,6 +555,40 @@ _public_ int sd_radv_add_prefix(sd_radv *ra, sd_radv_prefix *p) {
         return 0;
 }
 
+_public_ int sd_radv_set_rdnss(sd_radv *ra, uint32_t lifetime,
+                               struct in6_addr *dns, size_t dns_count) {
+        struct sd_radv_opt_rdnss *opt_rdnss;
+
+        assert_return(ra, -EINVAL);
+        assert_return(dns_count < 128, -EINVAL);
+
+        if (!dns || dns_count == 0) {
+                mfree(ra->rdnss);
+                ra->rdnss = NULL;
+                ra->n_rdnss = 0;
+
+                return 0;
+        }
+
+        opt_rdnss = malloc0(sizeof(struct sd_radv_opt_rdnss) +
+                            sizeof(struct in6_addr) * dns_count);
+        if (!opt_rdnss)
+                return -ENOMEM;
+
+        mfree(ra->rdnss);
+
+        ra->rdnss = opt_rdnss;
```

Comment:
Nitpicking myself: the above freeing of previous data and assignment of the new one should perhaps be done after memcpy below.